beacon-chain/forkchoice/doubly-linked-tree/proposer_boost.go

```diff
@@ -15,29 +16,44 @@ func (f *ForkChoice) ResetBoostedProposerRoot(_ context.Context) error {
 	return nil
 }
 
-// Given a list of validator balances, we compute the proposer boost score
-// that should be given to a proposer based on their committee weight, derived from
-// the total active balances, the size of a committee, and a boost score constant.
-// IMPORTANT: The caller MUST pass in a list of validator balances where balances > 0 refer to active
-// validators while balances == 0 are for inactive validators.
-func computeProposerBoostScore(validatorBalances []uint64) (score uint64, err error) {
-	totalActiveBalance := uint64(0)
-	numActive := uint64(0)
-	for _, balance := range validatorBalances {
-		// We only consider balances > 0. The input slice should be constructed
-		// as balance > 0 for all active validators and 0 for inactive ones.
-		if balance == 0 {
-			continue
+// applyProposerBoostScore applies the current proposer boost scores to the
+// relevant nodes. This function requires a lock in Store.nodesLock.
+func (f *ForkChoice) applyProposerBoostScore() error {
+	s := f.store
+	s.proposerBoostLock.Lock()
```

Comment:
its been a while since I looked at forkchoice code, but this mutex ordering won't cause any issues, correct ? I know that previously we had many deadlocks due to how the different locks/rlocks were acquired and released.

Reply:
Discussed offline